cert_issuer/blockchain_handlers/ethereum/signer.py

```diff
@@ -28,12 +27,13 @@ def sign_message(self, wif, message_to_sign):
     def sign_transaction(self, wif, transaction_to_sign):
         ##try to sign the transaction.
 
-        if isinstance(transaction_to_sign, transactions.Transaction):
+        if isinstance(transaction_to_sign, dict):
             try:
-                raw_tx = rlp.encode(transaction_to_sign.sign(wif, self.netcode))
-                raw_tx_hex = encode_hex(raw_tx)
+                transaction_to_sign['chainId'] = self.netcode
+                raw_tx = web3.Account.sign_transaction(transaction_to_sign, wif)['rawTransaction']
+                raw_tx_hex = to_hex(raw_tx)
                 return raw_tx_hex
             except Exception as msg:
                 return {'error': True, 'message': msg}
         else:
-            raise UnableToSignTxError('You are trying to sign a non transaction type')
+            raise UnableToSignTxError('sign_transaction expects a dict')
```

Comment:
could this error message be more helpful?

Reply:
OK, I'll rewrite it in just a bit